### PR TITLE
ffi: Support banning, unbanning and kicking users.

### PR DESCRIPTION
**Pull request by pixlwave**
`can_ban` and `can_kick` already exist and from what I can tell unbanning is covered by the `ban` power level.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 1]
```diff
@@ -424,6 +424,24 @@ impl Room {
         Ok(self.inner.can_user_ban(&user_id).await?)
     }
 
+    pub async fn ban_user(
```
[the remark this answers is not in the document]
Comment (Hywan) [reply bindings/matrix-sdk-ffi/src/room.rs:427]: The FFI API is designed for your needs mostly. If you believe it's fine as so, then I'm fine with it.

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 2]
```diff
@@ -424,6 +424,24 @@ impl Room {
...
+        &self,
+        user_id: String,
+        reason: Option<String>,
+    ) -> Result<(), ClientError> {
+        let user_id = UserId::parse(&user_id)?;
+        Ok(self.inner.ban_user(&user_id, reason.as_deref()).await?)
```
Comment (Hywan) [inline bindings/matrix-sdk-ffi/src/room.rs:433]: `as_deref` is nice here, well done! (I know you're learning Rust :-)).

File: crates/matrix-sdk/src/room/mod.rs [hunk 3]
```diff
@@ -1041,6 +1041,23 @@ impl Room {
...
+    /// * `reason` - The reason for unbanning this user.
+    #[instrument(skip_all)]
+    pub async fn unban_user(&self, user_id: &UserId, reason: Option<&str>) -> Result<()> {
+        let request = assign!(
+            unban_user::v3::Request::new(self.room_id().to_owned(), user_id.to_owned()),
+            { reason: reason.map(ToOwned::to_owned) }
+        );
+        self.client.send(request, None).await?;
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk/src/room/mod.rs:1059]: You can update this test case to test `unban_user` https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk/tests/integration/room/joined.rs#L109-L130.

**Review by Hywan — CHANGES_REQUESTED** [review]
Adding a test for `unban_user` would be very appreciated. Apart from that, the PR looks good, thanks!

File: bindings/matrix-sdk-ffi/src/room.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pixlwave) [reply bindings/matrix-sdk-ffi/src/room.rs:427]: Ok, then lets leave it like this for now. Can come back to this later if it makes sense in actual use.

**Review by bnjbvr — APPROVED** [review]

**Pull request merged**